src/lib/Inference.hs

```diff
@@ -1309,9 +1404,35 @@ inferUVar = \case
   UMethodVar v -> do
     MethodBinding _ _ f <- lookupEnv v
     return f
-  UEffectVar _ -> throw NotImplementedErr "inferUVar::UEffectVar"
-  UEffectOpVar _ -> throw NotImplementedErr "inferUVar::UEffectOpVar"
-  UHandlerVar _ -> throw NotImplementedErr "inferUVar::UHandlerVar"
+  UEffectVar _ -> do
+    throw NotImplementedErr "inferUVar::UEffectVar"  -- TODO(alex): implement
+  UEffectOpVar v -> do
```

Comment:
This reads to me a bit as if the effectful operations were first-class functions too. Are they like that? Is this allowed?
```
f = throw
case color of
  Red -> f ()
  Green -> f ()
  Blue -> 1.0
```

I thought that even any mention of `throw` would perform the throw operation

Reply:
I'm not sure why this would be a problem? Once everything gets inlined (which we're assuming will indeed happen), we would reduce to
```
case color of
  Red -> throw ()
  Green -> throw ()
  Blule -> 1.0
```
no?